spec/controllers/home_page_spec.rb

```diff
@@ -1,13 +1,28 @@
 require 'rails_helper'
 
 RSpec.describe HomePageController do
-  describe "GET #index" do
+  describe 'GET #index' do
     let(:do_request) { get :index }
 
-    it "renders the index template" do
-      do_request
-      expect(response).to have_http_status(:ok)
-      expect(response).to render_template(:index)
+    context 'not logged in user' do
+      it 'redirects to WelcomeController' do
+        do_request
+        expect(response).to have_http_status 302
+        expect(response).to redirect_to controller: 'welcome', action: 'index'
+      end
+    end
+
+    context 'logged in user' do
+
+      user = FactoryBot.create(:user)
```

Comment:
What do you think about this structure? I believe you've heard about BDD and there is a pattern which is appropriate to BDD. It called [Given-When-Then](https://martinfowler.com/bliki/GivenWhenThen.html). I like how `Let-Before-It` is similar to `Given-When-Then`
```ruby
context 'logged in user' do
  let(:user) { FactoryBot.create(:user) }

  before do
    sign_in user
    do_request
  end

  it 'renders the index template' do
    expect(response).to have_http_status(:ok)
    expect(response).to render_template(:index)
  end
end
```

Reply:
done